snooty/postprocess.py

```diff
@@ -902,20 +906,27 @@ def find_toctree_nodes(
 
                     toctree_node = {
                         "title": title,
-                        "slug": slug,
+                        "slug": "/" if slug == "index" else slug,
                         "children": [],
                         "options": {"drawer": slug not in self.toc_landing_pages},
                     }
 
-                    new_ast = self.pages[slug_fileid].ast
-                    self.find_toctree_nodes(slug_fileid, new_ast, toctree_node)
+                    # Don't recurse on the index page
+                    if slug_fileid not in visited_file_ids:
+                        new_ast = self.pages[slug_fileid].ast
+                        self.find_toctree_nodes(
+                            slug_fileid,
+                            new_ast,
+                            toctree_node,
+                            visited_file_ids | {slug_fileid},
```

Comment:
I think it may be a bit clearer to use the [`add()`](https://www.programiz.com/python-programming/methods/set/add) method instead, but your call!